instrumentation/apache-camel-2.20/javaagent-unittests/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/MessageAttributeNamesFactoryTest.groovy

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachecamel
+
+import io.opentelemetry.api.DefaultOpenTelemetry
+import io.opentelemetry.api.OpenTelemetry
+import io.opentelemetry.api.baggage.propagation.W3CBaggagePropagator
+import io.opentelemetry.context.propagation.ContextPropagators
+import io.opentelemetry.extension.trace.propagation.B3Propagator
+import org.apache.camel.component.aws.sqs.SqsConfiguration
+import org.apache.camel.component.aws.sqs.SqsEndpoint
+import spock.lang.Specification
+
+class MessageAttributeNamesFactoryTest extends Specification {
+
+  def "should Add Single Field To Empty Names List"() {
+
+    setup:
+    // traceparent = single field
+    OpenTelemetry.set(
+      DefaultOpenTelemetry.builder()
+        .setPropagators(ContextPropagators.create(W3CBaggagePropagator.getInstance())).build())
+
+    def sqsConfiguration = Mock(SqsConfiguration) {
+      getMessageAttributeNames() >> null
+    }
+    def sqsEndpoint = Mock(SqsEndpoint) {
+      getConfiguration() >> sqsConfiguration
+    }
+
+    when:
+    def names = MessageAttributeNamesFactory.withTextMapPropagatorFields(sqsEndpoint)
+
+    then:
+    names == "baggage"
+  }
+
+  def "should Add Single Field To Singleton Names List"() {
+    setup:
+    // traceparent = single field
```

Comment:
Why traceparent? You're not using the `W3CTraceContextPropagator` anywhere in this test